libs/ui-lib/lib/common/components/ui/formik/UploadField.tsx

```diff
@@ -36,64 +41,77 @@ const UploadField: React.FC<UploadFieldProps> = ({
     return '';
   };
   const errorMessage = getErrorMessage();
+  const acceptedFiles = { 'application/x-ssh-key': ['.pub'] };
+  const maxFileSize = 2048;
+  return (
+    <FormGroup fieldId={fieldId} label={label} isRequired={isRequired} labelIcon={labelIcon}>
+      {children}
+      <FileUpload
+        filenamePlaceholder={t('ai:Drag a file here or browse to upload')}
+        browseButtonText={t('ai:Browse...')}
+        clearButtonText={t('ai:Clear')}
+        id={field.name}
+        style={{ resize: 'vertical' }}
+        validated={isValid ? 'default' : 'error'}
+        isRequired={isRequired}
+        aria-describedby={`${fieldId}-helper`}
+        type="text"
+        value={field.value as string}
+        filename={filename}
+        onDataChange={(_event, file) => setValue(file)}
+        onTextChange={(_event, file) => setValue(file)}
+        onFileInputChange={(_event, file) => {
+          setFilename(file.name);
+          setTouched(true);
+          setValue(file);
+        }}
+        onBlur={(e) => {
+          const file: File | string = field.value;
+          if (file instanceof File) {
+            const reader = new FileReader();
 
-  const fieldHelperText = <HelperText fieldId={fieldId}>{helperText}</HelperText>;
+            reader.onload = (event) => {
+              if (event.target) {
+                const contentAsString = event.target.result as string;
+                setValue(trimSshPublicKey(contentAsString));
```

Comment:
`trimSshPublicKey` is particular to `UploadSSH` and shouldn't belong in this generic component.

Reply:
The question is that we use UploadField only inside UploadSSH. Maybe is better to have only one component?